### PR TITLE
Reorganize docs for easier deploys

### LINKED ISSUE #26267: Documentation / Components / Cards > Card columns >card text of last card doesn't fit
Card columns: Last Card: "... This card has even longer content than the first to show that equal height action." This cart-text belongs to Card-decks - in this place must be another (placeholder-) cart-text.

### LINKED ISSUE #26637: Twitter/Facebook Image not showing (broken image link)
I noticed that the image doesn't show in the Twitter Card when sharing a link to the Bootstrap Docs on Twitter. (e.g. https://twitter.com/christianoliff/status/1002077097844621312).

I see in the docs markup the following:
`<meta name="twitter:image" content="https://getbootstrap.com/assets/brand/bootstrap-social-logo.png">`
but https://getbootstrap.com/assets/brand/bootstrap-social-logo.png is broken link.

Same issue wtih Facebook sharing since the same image is referenced:
`<meta property="og:image" content="http://getbootstrap.com/assets/brand/bootstrap-social.png">`

Do the Twitter Card / OpenGraph image links need updating to the following?
`http://getbootstrap.com/docs/4.1/assets/brand/bootstrap-social.png`

### PR DESCRIPTION
**Pull request by mdo**
With the current docs directory setup, I'm making too many mistakes and have to manually address path changes and directory moves on deploy. This makes for a frustrating experience developing locally and shipping releases. With this PR, we're basically back to the same setup from v3—duplicating the `dist/` directory into our `site/docs/` directory. Not the most ideal, but very straightforward for me as the release manager.

I still need to double check the service worker stuff, asset paths, and confirm the watch tasks pickup the copy-pasta.

/cc @twbs/team 

---

- [x] `mkdir site`
- [x] `git mv docs/ site/docs/`
- [x] `git mv _layouts/ site/_layouts/`
- [x] `git mv _data/ site/_data/`
- [x] `git mv _includes site/_includes/`
- [x] `git mv assets site/docs/4.1/assets/`
- [x] Update `_config.yml`'s `source` to `"site"`
- [x] `git mv favicon.ico site/favicon.ico`
- [x] `git mv index.html site/index.html`
- [x] `git mv sw.js site/sw.js`
- [x] `git mv robots.txt site/robots.txt`
- [x] Update `sw.js` path
- [x] Update CSS/JS paths
- [x] Update social image paths
- [x] Modify `package.json` scripts do duplicate `dist` into `site` directory
	- [x] `css-main` needs ` && cp -r dist/css/ site/docs/4.1/dist/css/` added
	- [x] `js-main` needs ` && cp -r dist/js/ site/docs/4.1/dist/js/` added
- [ ] Test, test, test!

Fixes #26637 and fixes #26267 along the way.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Oops, sorry for the conflicts.

I haven't tested this myself, I will try it, but I'm not sure how many things one can catch locally :/ I mean, most, if not all of the issues, emerged in the live site.

**Comment by mdo** [comment]
> I haven't tested this myself, I will try it, but I'm not sure how many things one can catch locally :/ I mean, most, if not all of the issues, emerged in the live site.

That's right, and those all resulted from the versioning of our docs and me manually moving directories around and then globally find-and-replacing paths. Was bound to blow up in my face :). We'll know soon enough!

And no worries on conflicts, I can resolve.

**Comment by mdo** [comment]
@Johann-S @XhmikosR Can you peep the JS test changes in c38f49e? I'm not worried right now since these paths are referencing files that are not included in the npm package or anything like that.

**Comment by XhmikosR** [comment]
Looks good to me. Are there any other such changes in the branch? I still haven't had the time to test the branch :/ I'll do it later today.

**Comment by mdo** [comment]
@XhmikosR Pushed some more changes to fine tune the `watch` and `dist` scripts. This takes care of my two primary workflows, but I want to make sure paths, service worker, and other things are working for you locally with the latest changes here.

File: package.json [hunk 1]
```diff
@@ -98,13 +100,14 @@
     "bundlesize": "^0.15.3",
     "clean-css-cli": "^4.1.11",
     "coveralls": "^3.0.1",
+    "cpx": "^1.5.0",
```
Comment (XhmikosR) [inline package.json]: Why do we need cpx? We already have shx.

File: _config.yml [hunk 2]
```diff
@@ -9,33 +9,16 @@ kramdown:
...
-  - Gemfile
-  - Gemfile.lock
-  - package.js
-  - package.json
-  - package-lock.json
-  - README.md
-  - sache.json
-  - twbsconfig.yml
+  - docs/4.1/assets/scss/
+  - docs/4.1/assets/js/src/
```
Comment (XhmikosR) [inline _config.yml]: This is needed for development.

**Comment by XhmikosR** [comment]
Service worker generation is currently broken (`npm run docs-github`).

Also, I'm not sure the redirects make sense any more; we'd need to check each one.

And it seems one has to manually run the `css-copy` and `js-copy` tasks before building locally. You didn't notice because you have those files there already.

In that case, we are better off hooking up a `predocs-compile` script which will run the 2 copy tasks.

**Comment by mdo** [comment]
@XhmikosR For our service worker file, do we need to version that as well I assume? Would replacing it with v4.2's cause issues for the docs in `/4.1/` once live?

**Comment by XhmikosR** [comment]
I'm not sure about it, TBH. In theory, since the service worker is registered in the root, it will cover everything for us.

I still need to test this locally and try and see if there's something else broken.

**Comment by XhmikosR** [comment]
@mdo: can you do a rebase against v4-dev?

**Comment by mdo** [comment]
@XhmikosR I added some script changes to simplify local dev and more. Check out 21c3af9 and 492d3b5d3 when you have time—this should ensure the copy tasks happen at all times. I've tested changes to docs CSS and JS, as well as the main CSS and JS.

Also tested running the `npm start` to account for my usual dev workflow and running `build/ship.sh 4.1.2` to see how shipping goes. So far, so good, save for one thing in Chrome:

screen shot 2018-07-03 at 1 11 56 am

**Comment by XhmikosR** [comment]
@mdo: I'm gonna try to test this branch in the next days and let you know.

A couple of things:

1. tests break
2. it's the first time I see the inject.preload.js; make sure you clear the localhost service worker if you use the same address with a service worker. I will confirm this when I try the branch
3. should I update my rebased branch and force push to this one?

**Comment by XhmikosR** [comment]
Oh, might be one of your extensions for issue 2.

**Comment by mdo** [comment]
Can confirm, it's from an extension. Disabled them and it's clear. Broken tests appear to be from broken links—reviewing those now.

**Comment by XhmikosR** [comment]
@mdo: what should I do with https://github.com/twbs/bootstrap/compare/mv-docs-rebased?

**Comment by XhmikosR** [comment]
Also, @mdo do you want this in 4.1.2 or 4.2.0? If it's 4.1.2 I'll test it ASAP today or tomorrow.

**Review by XhmikosR — APPROVED** [review]

**Comment by mdo** [comment]
CI failures are very likely rate limits, so merging away!

**Pull request merged**